src/prefect/agent/kubernetes2/job_template.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: prefect-job
+  labels: {}
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: flow
+          image: prefecthq/prefect:latest
+          imagePullPolicy: IfNotPresent
+          command: ["/bin/sh", "-c"]
+          args: ["prefect execute flow-run"]
+          resources:
+            requests:
+              cpu: "100m"
```

Comment:
Independently of this PR, I think we should increase these defaults by a little

Reply:
What would you like them to be?

Reply:
cc @cicdw ^^

Reply:
I originally was thinking that we should move both the limit / request to something closer to 500m although I don't have a specific justification for that number; we've seen a few instances where this job is evicted / freezes and I suspect it's related to this low value.  

Because this is soon going to be _very_ easy to configure, I don't feel strongly about updating it immediately.

Reply:
I'm half tempted to remove it entirely. Imposing a limit (especially one this low) on a job by default is making assumptions for the user. Compute-heavy jobs will be heavily slowed by this limit. Most helm charts (at least the ones I've used) leave the resources field empty by default, leaving it up to the user to decide the resource requirements/limits.

My gut says that for naive users, having no request/limit here won't be a problem, and for expert/enterprise-y users, they'll likely want to configure the default template themselves anyway.

Not a strong thought though, up to you.

Reply:
just chiming in to say that this (0.1 limit) caused us quite a few hard-to-diagnose problems with the large mapping flow that we've discussed before...to me it really seems like the downsides to a default limit far outweigh the upsides

Reply:
In #3459 I propose we remove these limits (reasoning given in the PR description).